### PR TITLE
Proposed new 1.0.5 release.

### PR DESCRIPTION
**Pull request by genegr**
Fixed Prometheus configuration file for the docker example. Fixed issue #26

### CODE REVIEW

File: internal/openmetrics-exporter/pods_performance_replication_collector.go [hunk 1]
```diff
@@ -25,73 +25,73 @@ func (c *PodsPerformanceReplicationCollector) Collect(ch chan<- prometheus.Metri
 			c.BandwidthDesc,
 			prometheus.GaugeValue,
 		        p.ContinuousBytesPerSec.FromRemoteBytesPerSec,
-			"from_remote", "continuos", p.Pod.Name,
+			"continuos", "from_remote", p.Pod.Name,
```
Comment (sdodsley) [inline internal/openmetrics-exporter/pods_performance_replication_collector.go:28]: continuous

File: internal/openmetrics-exporter/pods_performance_replication_collector.go [hunk 2]
```diff
@@ -25,73 +25,73 @@ func (c *PodsPerformanceReplicationCollector) Collect(ch chan<- prometheus.Metri
...
 		        p.PeriodicBytesPerSec.FromRemoteBytesPerSec,
-			"from_remote", "periodic", p.Pod.Name,
+			"periodic", "from_remote", p.Pod.Name,
 		)
 		ch <- prometheus.MustNewConstMetric(
 			c.BandwidthDesc,
 			prometheus.GaugeValue,
 		        p.ContinuousBytesPerSec.ToRemoteBytesPerSec,
-			"to_remote", "continuos", p.Pod.Name,
+			"continuos", "to_remote", p.Pod.Name,
```
Comment (sdodsley) [inline internal/openmetrics-exporter/pods_performance_replication_collector.go:52]: continuous

File: internal/openmetrics-exporter/pods_performance_replication_collector.go [hunk 3]
```diff
@@ -25,73 +25,73 @@ func (c *PodsPerformanceReplicationCollector) Collect(ch chan<- prometheus.Metri
...
 		        p.PeriodicBytesPerSec.ToRemoteBytesPerSec,
-			"to_remote", "periodic", p.Pod.Name,
+			"periodic", "to_remote", p.Pod.Name,
 		)
 		ch <- prometheus.MustNewConstMetric(
 			c.BandwidthDesc,
 			prometheus.GaugeValue,
 		        p.ContinuousBytesPerSec.TotalBytesPerSec,
-			"total", "continuos", p.Pod.Name,
+			"continuos", "total", p.Pod.Name,
```
Comment (sdodsley) [inline internal/openmetrics-exporter/pods_performance_replication_collector.go:76]: continuous

File: internal/openmetrics-exporter/pods_performance_replication_collector_test.go [hunk 4]
```diff
@@ -37,19 +37,18 @@ func TestPodsPerformanceReplicationCollector(t *testing.T) {
...
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"continuos\" > label:<name:\"direction\" value:\"from_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ContinuousBytesPerSec.FromRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"resync\" > label:<name:\"direction\" value:\"from_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ResyncBytesPerSec.FromRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"sync\" > label:<name:\"direction\" value:\"from_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.SyncBytesPerSec.FromRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"periodic\" > label:<name:\"direction\" value:\"from_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.PeriodicBytesPerSec.FromRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"continuos\" > label:<name:\"direction\" value:\"to_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ContinuousBytesPerSec.ToRemoteBytesPerSec)] = true
```
Comment (sdodsley) [inline internal/openmetrics-exporter/pods_performance_replication_collector_test.go:44]: continuous

File: internal/openmetrics-exporter/pods_performance_replication_collector_test.go [hunk 5]
```diff
@@ -37,19 +37,18 @@ func TestPodsPerformanceReplicationCollector(t *testing.T) {
...
-
-		want[fmt.Sprintf("label:<name:\"dimension\" value:\"total\" > label:<name:\"direction\" value:\"continuos\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ContinuousBytesPerSec.TotalBytesPerSec)] = true
-		want[fmt.Sprintf("label:<name:\"dimension\" value:\"total\" > label:<name:\"direction\" value:\"resync\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ResyncBytesPerSec.TotalBytesPerSec)] = true
-		want[fmt.Sprintf("label:<name:\"dimension\" value:\"total\" > label:<name:\"direction\" value:\"sync\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.SyncBytesPerSec.TotalBytesPerSec)] = true
-		want[fmt.Sprintf("label:<name:\"dimension\" value:\"total\" > label:<name:\"direction\" value:\"periodic\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.PeriodicBytesPerSec.TotalBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"continuos\" > label:<name:\"direction\" value:\"from_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ContinuousBytesPerSec.FromRemoteBytesPerSec)] = true
```
Comment (sdodsley) [inline internal/openmetrics-exporter/pods_performance_replication_collector_test.go:40]: continuous

File: internal/openmetrics-exporter/pods_performance_replication_collector_test.go [hunk 6]
```diff
@@ -37,19 +37,18 @@ func TestPodsPerformanceReplicationCollector(t *testing.T) {
...
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"continuos\" > label:<name:\"direction\" value:\"to_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ContinuousBytesPerSec.ToRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"resync\" > label:<name:\"direction\" value:\"to_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ResyncBytesPerSec.ToRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"sync\" > label:<name:\"direction\" value:\"to_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.SyncBytesPerSec.ToRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"periodic\" > label:<name:\"direction\" value:\"to_remote\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.PeriodicBytesPerSec.ToRemoteBytesPerSec)] = true
+		want[fmt.Sprintf("label:<name:\"dimension\" value:\"continuos\" > label:<name:\"direction\" value:\"total\" > label:<name:\"name\" value:\"%s\" > gauge:<value:%g > ", p.Pod.Name, p.ContinuousBytesPerSec.TotalBytesPerSec)] = true
```
Comment (sdodsley) [inline internal/openmetrics-exporter/pods_performance_replication_collector_test.go:48]: continuous

**Review by sdodsley — CHANGES_REQUESTED** [review]

File: internal/openmetrics-exporter/pods_performance_replication_collector_test.go (hunk 5 above)
Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector_test.go]: fixed

File: internal/openmetrics-exporter/pods_performance_replication_collector.go (hunk 1 above)
Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector.go]: fixed

File: internal/openmetrics-exporter/pods_performance_replication_collector.go (hunk 2 above)
Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector.go]: fixed

File: internal/openmetrics-exporter/pods_performance_replication_collector.go (hunk 3 above)
Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector.go]: fixed

Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector.go]: fixed

Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector.go]: fixed

File: internal/openmetrics-exporter/pods_performance_replication_collector_test.go (hunk 6 above)
Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector_test.go]: fixed

**Review by sdodsley — APPROVED** [review]

File: go.mod [hunk 7]
```diff
@@ -18,6 +18,7 @@ require (
 	github.com/json-iterator/go v1.1.12 // indirect
 	github.com/kr/pretty v0.3.1 // indirect
 	github.com/matttproud/golang_protobuf_extensions v1.0.4 // indirect
+	//	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
```
Comment (chrroberts-pure) [inline go.mod:21]: this line can be removed

**Review by chrroberts-pure — DISMISSED** [review]
Looks good to me!

File: internal/openmetrics-exporter/pods_space_collector.go [hunk 8]
```diff
@@ -129,6 +144,12 @@ func NewPodsSpaceCollector(fa *client.FAClient) *PodsSpaceCollector {
 			[]string{"name", "space"},
 			prometheus.Labels{},
 		),
+		MediatorDesc: prometheus.NewDesc(
+			"purefa_pod_mediator_status",
+			"FlashArray pod space in bytes",
```
Comment (sdodsley) [inline internal/openmetrics-exporter/pods_space_collector.go:149]: This doesn't seem like the correct description

**Review by sdodsley — DISMISSED** [review]

File: Makefile [hunk 9]
```diff
@@ -4,7 +4,7 @@ GOTEST=$(GOCMD) test
 GOVET=$(GOCMD) vet
 BINARY_NAME=pure-fa-om-exporter
 MODULE_NAME=purestorage/fa-openmetrics-exporter
-VERSION?=1.0.4.hotfix1
+VERSION?=1.0.4.hotfix2
```
Comment (sdodsley) [inline Makefile:7]: change to 1.0.5

File: cmd/fa-om-exporter/main.go [hunk 10]
```diff
@@ -18,7 +18,7 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-var version string = "1.0.4.hotfix1"
+var version string = "1.0.4.hotfix2"
```
Comment (sdodsley) [inline cmd/fa-om-exporter/main.go:21]: 1.0.5 pleasae

**Review by sdodsley — CHANGES_REQUESTED** [review]

File: internal/openmetrics-exporter/pods_performance_replication_collector_test.go (hunk 4 above)
Comment (genegr) [reply internal/openmetrics-exporter/pods_performance_replication_collector_test.go]: fixed

File: go.mod (hunk 7 above)
Comment (genegr) [reply go.mod]: fixed

File: internal/openmetrics-exporter/pods_space_collector.go (hunk 8 above)
Comment (genegr) [reply internal/openmetrics-exporter/pods_space_collector.go]: changed with the proper description

File: Makefile (hunk 9 above)
Comment (genegr) [reply Makefile]: changed

File: cmd/fa-om-exporter/main.go (hunk 10 above)
Comment (genegr) [reply cmd/fa-om-exporter/main.go]: changed

**Review by sdodsley — APPROVED** [review]

**Review by chrroberts-pure — APPROVED** [review]
Approved as requested. Please just make sure to note the new metric names as [added] in the changelog.

also I would consider making this the 1.1.0 release since we are adding a large number of new metrics. Though, I see 1.0.5 was specifically requested. 

We should also consider in the future how we may be able to configure finer control over the metrics we collect (more so than /hosts, /array, /volumes...etc) However it is easy to set any o11y suite to ignore collected metrics.

**Pull request merged**